simulator/src/main/java/com/hazelcast/simulator/protocol/registry/AgentData.java

```diff
@@ -38,6 +39,7 @@
 public class AgentData {
 
     private final Collection<WorkerData> workers = new ArrayList<WorkerData>();
+    private final AtomicInteger currentWorkerIndex = new AtomicInteger();
```

Comment:
I would do the renaming in a different PR, since we use index all over the place. Actually we might run into trouble with it, since in the end `workerIndex` and `workerId` are already existing, where `workerIndex` is the numerical part (number, unique per Agent, increasing) and `workerId` is the complete name of the worker, including it's type (used e.g. as name for the `workerHome` directory).